### DISCUSSION TITLE
Engraving On curved surface

### DISCUSSION THREAD

**Discussion by troncatore** [arkypita/LaserGRBL#1382] [Q&A]
I would like to use my Naje laser master 2 for engraving curved surface...I made two my own A axe following CNC idea but it doesn't works properly due to on Grlb laser configuration Stepper on Y axe are calculated as linear movement. I posed the question to Naje and the answer was "On Grlb 1 step is 0,025 mm" meaning unuseful answer. Does anyone are using Y axe as a rotation axe? thanks for answering

**Comment by arkypita** [comment]
I don't know if it help but usually curved surface are done with rotary roller.
![image](https://user-images.githubusercontent.com/8782035/112126069-5f151480-8bc4-11eb-97cd-404baec51e76.png)

A rotary roller transform linear to rotative movement without need to keep in mind of piece diameter or other stuff. Just engrave like if it was linear.

If you have a lathe-style rotary axis you have to compute step based on diameter.

**Comment by troncatore** [comment]
Yes I know I did two arrangement for it it following a thingiverse solution...but the problem is using Y axe as a rotation...which is the correct setting step/degree instead step/millimeters?Thanks for answering
    Il martedì 23 marzo 2021, 10:44:53 CET, arkypita ***@***.***> ha scritto:  
 
 


I don't know if it help but usually curved surface are done with rotary roller.


A rotary roller transform linear to rotative movement without need to keep in mind of piece diameter or other stuff. Just engrave like if it was linear.

If you have a lathe-style rotary axis you have to compute step based on diameter.

**Reply by Troncatore1** [reply]
Ciao... Non riesco ad entrare sul mio vecchio account e ne ho fatto un'altro... Ora ho acquistato un AtomStack R1 e ho il problema dei step /mm... Come si calcolano in base al diametro del pezzo?
